hatchways/team-bluejay
# Moshe sockets for receiving notifications #61

See Ticket #61 

Description:

- Added notifications icon on front end navbar that opens a list of notifications when clicked
- Added socket.io to keep track of which users are online
- Created backend end function `notifyUser` to easily send notifications to users
- Chefs receive a notification when a customer places an order with them
- If chef is online they will receive new notifications via socketio immediately 

To test: 

- Run `python db_seeder.py`
- Login as `chef1@chef.com` with password `p@ssword1`. Verify you have two unread and two read notifications. Click outside notification menu and verify all four notifications now show as read.
- Login as a chef on one browser and on another browser as a customer. (one browser should be incognito mode)
- The customer should place an order with the chef. Verify chef receives the notification immediately in real-time (via socketio). Notification should initially be unread bold text. After closing notification menu and reopening the notification text should be normal read text.
- Log out as the chef.  Place another order as customer. Log back in as chef. Verify the new notification is there.
